internal/bridge/crunchybridgecluster/crunchybridgecluster_controller.go

```diff
@@ -328,9 +330,53 @@ func (r *CrunchyBridgeClusterReconciler) Reconcile(ctx context.Context, req ctrl
 
 		for _, cluster := range clusters {
 			if crunchybridgecluster.Name == cluster.Name {
-				crunchybridgecluster.Status.ID = cluster.ID
-				// Requeue now that we have a cluster ID assigned
-				return ctrl.Result{Requeue: true}, nil
+				// Cluster with the same name exists so check for adoption annotation
+				adoptionID, annotationExists := crunchybridgecluster.Annotations[naming.CrunchyBridgeClusterAdoptionAnnotation]
+				if strings.EqualFold(adoptionID, cluster.ID) {
+					// Annotation is present with correct ID value; adopt cluster by assigning ID to status.
+					crunchybridgecluster.Status.ID = cluster.ID
+					// Requeue now that we have a cluster ID assigned
+					return ctrl.Result{Requeue: true}, nil
+				} else {
+					// Adoption annotation value is incorrect. Check to see if annotation exists
+					// to determine proper log message and status condition
+					if annotationExists {
```

Comment:
🤔 This branch doesn't seem necessary to me. The longer message on L360 has the information from the API call. If the user can see `status` they can also see `annotations` and the mismatch.